src/device/device_cold.hpp

```diff
@@ -65,6 +115,54 @@ namespace hw {
                  const ::tools::wallet2::unsigned_tx_set & unsigned_tx,
                  ::tools::wallet2::signed_tx_set & signed_tx,
                  tx_aux_data & aux_data) =0;
+
+    /**
+     * Get tx key support check.
+     */
+    virtual bool is_get_tx_key_supported() const { return false; }
+
+    /**
+     * Loads TX aux data required for tx key.
+     */
+    virtual void load_tx_key_data(tx_key_data_t & res, const std::string & tx_aux_data) =0;
+
+    /**
+     * Decrypts TX keys.
+     * If view_public_key is set, derivations are computed instead of the tx private keys.
```

Comment:
While they're the same size, key derivations would be better stored in a crypto::key_derivation, or it's doing to be massively confusing. Might be better to have two functions in this case (maybe calling the same internal code, I've not seen it yet).

Reply:
True. I've removed this part and kept just tx_key computation as I noticed the intended use-case for derivations would require a bit change (tx_proof) which is not yet implemented in Trezor.